### ISSUE TITLE
Ignore files in `.gitignore` when auto-detecting monorepo projects

### ISSUE THREAD

**Issue by f-f** [purescript/spago#918]
Bug report from @thomashoneyman:
> when you use NPM workspaces, the contents of each workspace member are symlinked into the `node_modules` directory. That means the `spago.yaml` file is, too. That causes Spago to auto-detect the project within node_modules and try to build it as another workspace member (in this case, shadowing my actual one).

A good solution for this would be to not look for projects in places that are `gitignore`d.
This is where we'd need to add paths to exclude:

https://github.com/purescript/spago/blob/e611cace9811a433eba275235a8b9901dfc8981d/spaghetto/src/Spago/Config.purs#L302

**Comment by f-f** [comment]
Fixed in https://github.com/purescript/spago/commit/788393114efad97e5fd980b9cd09ce830b587241

**Comment by f-f** [comment]
I'm not sure we released a new version with this fix! Not at a computer right now, I can check later

**Comment by thomashoneyman** [comment]
Oh! Sorry, I thought it was present in 0.91.0 -- in that case I take it back!

**Comment by f-f** [comment]
It looks like we did - so we are probably just missing some detail.
@thomashoneyman if you run with `-v`, the debug logging will tell you which Spago configs it auto-discovered - what do you see there?

**Comment by thomashoneyman** [comment]
```
Reading Spago workspace configuration...
Reading config from spago.yaml
Found packages at these paths:
  app/spago.yaml
  lib/spago.yaml
  scripts/spago.yaml
  node_modules/registry-app/spago.yaml
  node_modules/registry-lib/spago.yaml
```

**Comment by f-f** [comment]
Uh, I can repro this. I'll have a look..

**Comment by f-f** [comment]
(in all of this, a `spago build` fails with a mysterious error)

**Comment by f-f** [comment]
Ah, symbolic links always throw a wrench in:
```
fatal: pathspec 'node_modules/registry-app/spago.yaml' is beyond a symbolic link
```

I'll cook a patch, but to use it in the registry we'll have to adjust the `spago run` cwd thing, as that has been fixed too

**Comment by f-f** [comment]
Fixed again in `0.91.1`
